db/migrate/20150716163704_remove_filesystem_repositories.rb

```diff
@@ -30,12 +30,37 @@
 # Removes all remaining Repository::Filesystem entries.
 #
 class RemoveFilesystemRepositories < ActiveRecord::Migration
+  include Migration::Utils
+
   def up
+    # Delete any changesets belonging to filesystem repositories
+    if postgres?
+      ActiveRecord::Base.connection.execute <<-SQL
+        DELETE FROM changesets
+        WHERE repository_id IN (
+          SELECT id
+          FROM repositories
+          WHERE type = #{filesystem_type}
+        )
```

Comment:
I suppose it would. I initially wanted to use Postgres' `USING` but then left this in. subqueries with IN are usually slower , even though it should not matter here. If you want me to change the migration, I'd switch to `USING`, which internally joins the repositories table.